Mathlib/RingTheory/Ideal/Maps.lean

```diff
@@ -0,0 +1,913 @@
+/-
+Copyright (c) 2018 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+import Mathlib.RingTheory.Ideal.Operations
+
+#align_import ring_theory.ideal.operations from "leanprover-community/mathlib"@"e7f0ddbf65bd7181a85edb74b64bdc35ba4bdc74"
+
+/-!
+# Maps on modules and ideals
+-/
+
+assert_not_exists Basis -- See `RingTheory.Ideal.Basis`
+assert_not_exists Submodule.hasQuotient -- See `RingTheory.Ideal.QuotientOperations`
+
+universe u v w x
+
+open BigOperators Pointwise
+
+namespace Ideal
+
+section MapAndComap
+
+variable {R : Type u} {S : Type v}
+
+section Semiring
+
+variable {F : Type*} [Semiring R] [Semiring S]
+variable [FunLike F R S] [rc : RingHomClass F R S]
+variable (f : F)
+variable {I J : Ideal R} {K L : Ideal S}
+
+/-- `I.map f` is the span of the image of the ideal `I` under `f`, which may be bigger than
+  the image itself. -/
+def map (I : Ideal R) : Ideal S :=
+  span (f '' I)
+#align ideal.map Ideal.map
+
+/-- `I.comap f` is the preimage of `I` under `f`. -/
+def comap (I : Ideal S) : Ideal R where
+  carrier := f ⁻¹' I
+  add_mem' {x y} hx hy := by
+    simp only [Set.mem_preimage, SetLike.mem_coe, map_add f] at hx hy ⊢
+    exact add_mem hx hy
+  zero_mem' := by simp only [Set.mem_preimage, map_zero, SetLike.mem_coe, Submodule.zero_mem]
+  smul_mem' c x hx := by
+    simp only [smul_eq_mul, Set.mem_preimage, map_mul, SetLike.mem_coe] at *
+    exact mul_mem_left I _ hx
+#align ideal.comap Ideal.comap
+
+-- Porting note (#10756): new theorem
```

Comment:
I think you might as well drop that porting note

Reply:
I also addressed the note on the next line
```lean
-- @[simp] -- Porting note (#11215): TODO enable simp after the port
```
and enabled the simp. Let's see if that works...

Reply:
Looks fine.